examples/attestation/main.go

```diff
@@ -0,0 +1,308 @@
+// Copyright 2023 Mariano Cano
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"log"
+	"math/big"
+	"net/http"
+	"os"
+
+	"github.com/mholt/acmez"
+	"github.com/mholt/acmez/acme"
+	"go.uber.org/zap"
+)
+
+const usage = `Usage: STTY=-icanon attestation <csr-file>
+
+<csr-file> A file with a certificate signing request or CSR.
+
+To be able to run this example, we need to use a key that can be attested,
+"step-ca" [1], for example, supports attestation using YubiKey 5 Series.
+
+To configure "step-ca" with device-attest-01 support, you need to create an ACME
+provisioner with the devices-attest-01 challenge enabled, in the ca.json the
+provisioner looks like this:
+
+  {
+    "type": "ACME",
+    "name": "attestation",
+    "challenges": [ "device-attest-01" ]
```

Comment:
Apple, step, and TPM are enabled by default.

Reply:
Yes, that's why I mentioned it. Do we want just `step`, or are the defaults OK? I think explicit is better, so that not just any iOS device can get a cert. It's just an example, but people might see and use it as a base.